### PR TITLE
Refactor implementations of AbstractExecutionWrapper

### PR DESCRIPTION
**Pull request by ashamis**
These classes used a pattern of have std::functions, this just not necessary anymore. Also from a previous discussion rename some parameters.

### CODE REVIEW

File: src/kv/deserialise.h [hunk 1]
```diff
@@ -238,14 +205,14 @@ namespace kv
...
 
-    Tx& get_tx() override
+    aft::Request& get_request() override
     {
-      return *tx;
+      return req;
     }
 
-    kv::Version get_max_conflict_version() override
+    virtual kv::Version get_max_conflict_version() override
```
Comment (achamayou) [inline src/kv/deserialise.h:213]:
```suggestion
    kv::Version get_max_conflict_version() override
```

File: src/kv/deserialise.h [hunk 2]
```diff
@@ -474,32 +370,11 @@ namespace kv
...
-             const std::vector<uint8_t>& data,
-             std::shared_ptr<TxHistory> history,
-             std::shared_ptr<ccf::ProgressTracker> progress_tracker,
-             kv::Version& v,
-             OrderedChanges& changes,
-             MapCollection& new_maps,
-             kv::ConsensusHookPtrs& hooks) -> ApplyResult {
       if (!store->commit_deserialised(changes, v, new_maps, hooks))
       {
+        LOG_FAIL_FMT("receive_nonces commit_deserialized Failed");
```
Comment (achamayou) [inline src/kv/deserialise.h:377]:
```suggestion
        LOG_FAIL_FMT("receive_nonces commit_deserialised Failed");
```

Not a comment on US vs UK spelling, just consistency with line :375.

**Review by achamayou — APPROVED** [review]

**Review by achamayou — APPROVED** [review]

**Comment by achamayou** [comment]
@ashamis can we repeat the perf test to check the ls_sgx_cft impact is just noise?

File: src/kv/deserialise.h [hunk 3]
```diff
@@ -623,23 +460,34 @@ namespace kv
...
+    }
+
+    kv::Version get_max_conflict_version() override
+    {
+      return max_conflict_version;
+    }
+
+  private:
+    uint64_t max_conflict_version;
+    std::unique_ptr<Tx> tx;
```
Comment (eddyashton) [inline src/kv/deserialise.h:491]: Please move definition of these members to the top of the class rather than the bottom.

**Comment by ashamis** [comment]
as per @achamayou comment, i reran the perf tests and the issue is transient

File: src/kv/kv_types.h [hunk 4]
```diff
@@ -576,7 +581,7 @@ namespace kv
     virtual std::shared_ptr<Consensus> get_consensus() = 0;
     virtual std::shared_ptr<TxHistory> get_history() = 0;
     virtual EncryptorPtr get_encryptor() = 0;
-    virtual std::unique_ptr<AbstractExecutionWrapper> apply(
+    virtual std::unique_ptr<AbstractExecutionWrapper> deserialize(
       const std::vector<uint8_t> data,
```
Comment (eddyashton) [inline src/kv/kv_types.h:585]: I think this arg should be `const&` to avoid an unnecessary copy.

**Pull request merged**